spec/options/strictNumbers.spec.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var Ajv = require('../ajv');
+
+describe('structNumbers option', function() {
+  var ajv;
+  describe('strictNumbers default', testWithoutStrictNumbers(new Ajv()));
+  describe('strictNumbers = false', testWithoutStrictNumbers(new Ajv({strictNumbers: false})));
+  describe('strictNumbers = true', function() {
+    beforeEach(function () {
+      ajv = new Ajv({strictNumbers: true});
+    });
+
+    it('should fail validation for NaN/Infinity as type number', function() {
+      var validate = ajv.compile({type: 'number'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(false);
+    });
+
+    it('should not impact validation for NaN as type integer', function() {
+      var validate = ajv.compile({type: 'integer'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(true);
```

Comment:
Just noticed it, sorry - I think integer should also not allow Infinity with strictNumbers.

Reply:
I don't think anybody would want to have a separate option and integers are also numbers (i.e. they would not fail validation with type: 'number'). The idea of type: 'integer' is to impose stricter constraint than number, but in this case it becomes somewhat overlapping rather than stricter.

Reply:
And I would also add non-integer examples to the tests, to make sure nothing was broken with the new option (e.g. 1.1 would pass number but fail integer).